### PR TITLE
Bulk settlement report [2/4]

### PR DESCRIPTION
**Pull request by KrekoG**
## Added
- Option to export aggregate settlement report for a period (#157183787)

[Delivers [#157183787](https://www.pivotaltracker.com/story/show/157183787)]

### CODE REVIEW

**Comment by KrekoG** [comment]
@00eadeyemi Ready for test review, don't forget it also requires the sdk branch `aggregate`

**Comment by 00eadeyemi** [comment]
👍 All tested

File: app/Http/Controllers/SettlementsController.php [hunk 1]
```diff
@@ -102,6 +102,40 @@ public function index(Merchant $merchant, Request $request)
...
+                $this
+                    ->settlementGateway
+                    ->getAggregateSettlementReports(
+                        $merchant->token,
+                        $dateRange['date_from'],
+                        $dateRange['date_to']
+                    )
+            );
+        } catch (\Exception $e) {
+            if ($e->getMessage() === 'Date range provided is invalid') {
```
Comment (EvanBarbour3) [inline app/Http/Controllers/SettlementsController.php]: Meh but okay

File: resources/assets/js/components/settlements/ExportButton.vue [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        beforeMount() {
+            this.$bus.$on('exportDateUpdate', this.exportDateUpdate);
+        },
+        beforeDestroy() {
+            this.$bus.$off('exportDateUpdate', this.exportDateUpdate);
+        },
+    }
+</script>
+
+<style scoped>
```
Comment (EvanBarbour3) [inline resources/assets/js/components/settlements/ExportButton.vue]: 👍

**Pull request merged**